### PR TITLE
Useful button

### LINKED ISSUE #1072: [new feature] useful button
### Component infos

A "useful" button in the description of a how-to page.

### Description
This button will allow users to give an appreciation on the content available in the how-to.

- The button is always visible to all users.
- Only logged in users can click it and increment it's value
- If user not logged in the button is disabled state and you get a message on hover (see mockup section of this issue)
- Every user can only give one “useful” per page 
- Clicking the button again removes the “useful”
- The amount of useful count is displayed in the button

### Page related
Will be used in :

- How-to page

### Mockup
Capture d’écran 2020-11-05 à 18 02 52

Capture d’écran 2020-11-05 à 18 09 43

Capture d’écran 2020-11-05 à 18 11 13

### Build suggestion
- [ ] Display the button on desktop and mobile. Color code of button background is `f5ede2`. Use [react-icons](https://react-icons.github.io) for the star icon.
Note : it will probably be in conflict with the admin moderation buttons. Align the useful button as on the mockup and align admin moderation buttons on the right on desktop. On mobile align the admin moderation buttons on the next line.
- [ ] link with database : add a `usefulCount` field on how-to model incremented or decremented on click. We should also create an array with the list of how-to a user as marked as "useful" in `userStats` collection.
- [ ] button display logic : clickable only for logged in users & display sign in message for not logged in users. Also display `usefulCount` in button if present.

If more questions feel free to ask them in this issue thread 🖖

### PR DESCRIPTION
**Pull request by tudi2d**
PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)
- [x] - Passes Tests

## Description

New "useful" button to give easy feedback. With this button there are 2 new fields added to the Firestore.

1. usefulCount: number (v3_howtos)
2. usefulHowTos: string[] (v3_users)

If a user marks a HowTo as useful for the first time the "usefulHowTos" field is added to the user profile and the "usefulCount"-field is added to the HowTo. This way everything should function as normal and there is no need of touching the database.

For not logged in users a Box is displayed with the same content as the Button, but - of course - not clickable.

Please test this for different edge cases as I´m not aware of all possible use cases and what rights a user can have.

## Git Issues

Closes #1072 

## Screenshots/Videos
image

zn1MpwwTui

### CODE REVIEW

File: src/models/user.models.tsx [hunk 1]
```diff
@@ -58,6 +58,7 @@ interface IExternalLink {
 interface IUserStats {
   userCreatedHowtos: { [id: string]: IModerationStatus }
   userCreatedEvents: { [id: string]: IModerationStatus }
+  usefUlHowTos: string[]
```
Comment (BenGamma) [inline src/models/user.models.tsx:61]: Change capital U here, it should be `usefulHowTos`

**Review by BenGamma — CHANGES_REQUESTED** [review]
Thanks man for the work 💪 
I noticed there are a lot of test failing, probably due to the fact that the new button is not expected. Can you have a look at the tests and fix them ? There is a section in [the wiki](https://github.com/ONEARMY/community-platform/wiki/E2E-Testing) about it

**Comment by tudi2d** [comment]
@BenGamma Thanks! Yes, sure - will fix the failing test as soon as everything is implemented and change the status from draft to PR then! Should find time for this in the next days! :)

**Review by chrismclarke — COMMENTED** [review]
Thanks again for this @tudi2d and sorry it's taken so long to review.

I really like the new feature overall, especially the separation between users recording on their own profile and just reflecting the total value on the howto. I've taken this one step further and actually moved that second piece of logic to a backend function to be triggered automatically (that way if a user disconnects or multiple users try to update at the same time data won't get lost).

I also realised one fundamental issue with how firestore operates, meaning that we will need to rethink how we display the final total. I've created a new PR with your changes and my edits on top at #1094, alongside more explanation of the changes (specifically c74562496580c95d0a4fe25c5be4a483a2654894)

**Comment by chrismclarke** [comment]
merging into #1094
